### PR TITLE
Improve COCO results (FPN + NMS changes).

### PR DESCRIPTION
**Pull request by hgaiser**
This PR does the following:

- Correct FPN computation as per their paper.
- Correct bbox transform formula (https://github.com/fizyr/keras-retinanet/issues/171).
- Change default image size to 800px.
- Change NMS to do per class NMS.

Since this is a breaking change for all models trained before this PR, I propose to tag `master` as 0.2 after merging and publishing a new pretrained model.

### CODE REVIEW

File: keras_retinanet/utils/anchors.py [hunk 1]
```diff
@@ -171,16 +171,20 @@ def bbox_transform(anchors, gt_boxes, mean=None, std=None):
     elif not isinstance(std, np.ndarray):
         raise ValueError('Expected std to be a np.ndarray, list or tuple. Received: {}'.format(type(std)))
 
-    anchor_widths  = anchors[:, 2] - anchors[:, 0] + 1.0
-    anchor_heights = anchors[:, 3] - anchors[:, 1] + 1.0
+    anchor_widths  = anchors[:, 2] - anchors[:, 0]
```
Comment (awilliamson) [inline keras_retinanet/utils/anchors.py]: I thought we needed these +1.0s due to dataset reasons? Can't fully recall, but I remember being told something along those lines?

File: keras_retinanet/models/retinanet.py [hunk 2]
```diff
@@ -103,14 +103,15 @@ def default_regression_model(num_anchors, pyramid_feature_size=256, regression_f
 
 def __create_pyramid_features(C3, C4, C5, feature_size=256):
     # upsample C5 to get P5 from the FPN paper
-    P5           = keras.layers.Conv2D(feature_size, kernel_size=1, strides=1, padding='same', name='P5')(C5)
+    P5           = keras.layers.Conv2D(feature_size, kernel_size=1, strides=1, padding='same', name='C5_reduced')(C5)
```
Comment (awilliamson) [inline keras_retinanet/models/retinanet.py]: Verified this is in line with Faster RCNN paper and their Pyramid outline, including the exceptions noted in the Focal Loss paper. This was done at the time @hgaiser was writing.

File: keras_retinanet/utils/anchors.py (hunk 1 above)
Comment (hgaiser) [reply keras_retinanet/utils/anchors.py]: Those +1's are from `py-faster-rcnn` originally. They encode a box as `[x1, x2]`, `[y1, y2]`, we encode it as `[x1, x2)` and `[y1, y2)`. That means we should compute the width as `x2 - x1` and height as `y2 - y1`. `py-faster-rcnn` (and Detectron for that matter) computes it as `x2 - x1 + 1` and `y2 - y1 + 1`.

Somewhere a while back we switched to this new notation, which in my opinion makes much more sense. However this code was left unchanged.
